happi/cli.py

```diff
@@ -121,8 +121,7 @@ def search(
                 range_list = new_range_list
             else:
                 # subsequent ranges, only take intersection
-                final_range = [item for item in new_range_list
-                               if item in range_list]
+                final_range = set(new_range_list) & set(range_list)
                 range_list = final_range
```

Comment:
Probably should have caught this last PR: this bit can be dropped down to one line, right? I don't think `final_range` is ever used, except to be set here.